src/NUnitFramework/framework/Internal/TestNameGenerator.cs

```diff
@@ -220,7 +220,47 @@ protected static string GetDisplayString(object arg, int stringMax)
                     ? "null"
                     : Convert.ToString(arg, System.Globalization.CultureInfo.InvariantCulture);
 
-                if (arg is double)
+                var argArray = arg as Array;
+                if (argArray != null)
+                {
+                    if (argArray.Length == 0)
+                        display = "[]";
+                    else
+                    {
```

Comment:
Lol, braces on the `else` but not the `if` always made me feel unbalanced. But I don't see anything wrong with it.